Lib/asyncio/tasks.py

```diff
@@ -432,8 +434,8 @@ async def wait_for(fut, timeout):
         # wait until the future completes or the timeout
         try:
             await waiter
-        except exceptions.CancelledError:
-            if fut.done():
+        except exceptions.CancelledError as e:
+            if fut.done() and _SENTINEL in e.args:
```

Comment:
By checking for a sentinel, we can suppress a `CancelledError` caused by us, while ignoring explicit cancellations caused outside of this code.